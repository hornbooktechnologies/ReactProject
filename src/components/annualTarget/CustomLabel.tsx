import React from 'react'
import { ChartConstant } from "../../utils/AppConstants";
import { useTranslation } from 'react-i18next'

const CustomLabel = (props: { viewBox: { x: number; y: number; }; }) => {
  const { i18n } = useTranslation();
  return (
    <>
     <g fill={ChartConstant.LABEL_COLOR} transform="rotate(-90)">
        <rect
          y={props.viewBox?.x - ChartConstant.VIEWBOX_20}
          x={props.viewBox?.y - ChartConstant.VIEWBOX_75}
          fill={ChartConstant.LABEL_COLOR}
          rx="2"
          width={50}
          height={20}
        />
        <text
          className="custom-label-text"
          y={props.viewBox?.x - ChartConstant.VIEWBOX_20}
          x={props.viewBox?.y - ChartConstant.VIEWBOX_75}
          radius={10}
          fill={ChartConstant.LABEL_TEXT_COLOR}
          dy={13}
          dx={10}
        >
          {i18n.t('LABELS.PRESENT')}
        </text>
      </g>
    </>
  );
};
export default CustomLabel;