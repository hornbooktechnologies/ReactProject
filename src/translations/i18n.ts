import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
