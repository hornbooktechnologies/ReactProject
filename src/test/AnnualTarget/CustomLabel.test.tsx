import React from "react";
import CustomLabel from "../../components/annualTarget/CustomLabel";
import renderer from "react-test-renderer";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";
describe("Component: CustomLabel", () => {
  it("CustomLabel renders correctly", () => {
    const tree = renderer.create(
      <I18nextProvider i18n={i18n}>
    <CustomLabel />
    </I18nextProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
