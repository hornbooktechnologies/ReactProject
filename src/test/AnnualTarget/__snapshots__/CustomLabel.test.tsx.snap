// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CustomLabel CustomLabel renders correctly 1`] = `
<g
  fill="#AAB9C1"
  transform="rotate(-90)"
>
  <rect
    fill="#AAB9C1"
    height={20}
    rx="2"
    width={50}
    x={NaN}
    y={NaN}
  />
  <text
    className="custom-label-text"
    dx={10}
    dy={13}
    fill="#231F20"
    radius={10}
    x={NaN}
    y={NaN}
  >
    LABELS.PRESENT
  </text>
</g>
`;
